test/libp2p_secure_framed_stream_SUITE.erl

```diff
@@ -0,0 +1,236 @@
+init_per_testcase(_, _Config) ->
+    test_util:setup(),
+    lager:set_loglevel(lager_console_backend, info),
+    _Config.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%   Special end config for test case
+%% @end
+%%--------------------------------------------------------------------
+end_per_testcase(_, _Config) ->
+    ok.
+
+%%--------------------------------------------------------------------
+%% TEST CASES
+%%--------------------------------------------------------------------
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+invalid_key_exchange(_Config) ->
+    SwarmOpts = [{libp2p_nat, [{enabled, false}]}],
+    Version = "securetest/1.0.0",
+
+    {ok, ServerSwarm} = libp2p_swarm:start(insecure_server_test, SwarmOpts),
+    ok = libp2p_swarm:listen(ServerSwarm, "/ip4/0.0.0.0/tcp/0"),
+    libp2p_swarm:add_stream_handler(
+        ServerSwarm,
+        Version,
+        {libp2p_framed_stream, server, [libp2p_secure_framed_stream_echo_test, self(), ServerSwarm, {secured, ServerSwarm}]}
+    ),
+
+    {ok, ClientSwarm} = libp2p_swarm:start(insecure_client_test, SwarmOpts),
+    ok = libp2p_swarm:listen(ClientSwarm, "/ip4/0.0.0.0/tcp/0"),
+
+    LAs = libp2p_swarm:listen_addrs(ServerSwarm),
+    [ServerAddress|_] = LAs,
+    ServerP2P = libp2p_swarm:p2p_address(ServerSwarm),
+
+    % Dialing here to propagate peerbook
+    {ok, ClientStream0} = libp2p_swarm:dial_framed_stream(
+        ClientSwarm,
+        ServerAddress,
+        Version,
+        libp2p_secure_framed_stream_echo_test,
+        [ClientSwarm, self()]
+    ),
+
+    PeerBook = libp2p_swarm:peerbook(ClientSwarm),
+    ServerPeerbookAddr = libp2p_swarm:pubkey_bin(ServerSwarm),
+    timer:sleep(1000),
+
+    ok = test_util:wait_until(fun() ->
+                                      ok == element(1, libp2p_peerbook:get(PeerBook, ServerPeerbookAddr))
+                              end),
+
+    gen_server:stop(ClientStream0),
+
+    ok = libp2p_swarm:stop(ServerSwarm),
+
+    %% start another swarm on the same address
+    {ok, EvilSwarm} = libp2p_swarm:start(insecure_evil_test, SwarmOpts),
+    [ok = libp2p_swarm:listen(EvilSwarm, LA) || LA <- LAs],
+    libp2p_swarm:add_stream_handler(
+        EvilSwarm,
+        Version,
+        {libp2p_framed_stream, server, [libp2p_secure_framed_stream_echo_test, self(), EvilSwarm, {secured, EvilSwarm}]}
+     ),
+
+    %% this should fail because the swarm on the other end is not who we think it is
+    ?assertEqual({error, incorrect_peer}, libp2p_swarm:dial_framed_stream(
+        ClientSwarm,
+        ServerP2P,
+        Version,
+        libp2p_secure_framed_stream_echo_test,
+        [ClientSwarm, self(), {secured, ClientSwarm}]
+    )),
+
+    ok = libp2p_swarm:stop(EvilSwarm),
+    ok = libp2p_swarm:stop(ClientSwarm),
+    ok.
+
+key_exchange(_Config) ->
```

Comment:
So is this test for the old 'compatible' libp2p secure streams that we abandoned? If so should we remove it?